interfaces/builtin/nvidia_tegra_drivers_support.go

```diff
@@ -0,0 +1,175 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2024 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+const nvidiaTegraDriversSupportSummary = `allows hardware access to NVIDIA tegra platforms`
+
+const nvidiaTegraDriversSupportBaseDeclarationSlots = `
+  nvidia-tegra-drivers-support:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: true
+`
+
+const nvidiaTegraDriversSupportConnectedPlugAppArmor = `
+# This is inverse of
+# https://forum.snapcraft.io/t/call-for-testing-chromium-62-0-3202-62/2569/46
+# As nvidia-assemble snap needs to create the static & dynamic MAJOR
+# chrdevs for all other snaps to have access to. Specifically
+/{,usr/}bin/mknod ixr,
+allow capability mknod,
+
+# To read dynamically allocated MAJOR for nvidia-uvm
+@{PROC}/modules r,
+@{PROC}/devices r,
+@{PROC}/driver/nvidia/params r,
+@{PROC}/driver/nvidia/capabilities/mig/monitor r,
+@{PROC}/driver/nvidia/capabilities/mig/config r,
+@{PROC}/sys/vm/mmap_min_addr r,
+
+/sys/devices/soc0/platform r,
+/sys/devices/soc0/soc_id r,
+/sys/devices/soc0/revision r,
+/sys/devices/soc0/major r,
+/sys/devices/platform/bus@0/3810000.fuse/fuse/nvmem r,
+
+/dev/nvmap rw,
+/dev/dri/renderD128 rw,
+/dev/nvgpu/igpu0/power rw,
+/dev/nvgpu/igpu0/ctrl rw,
+/dev/nvgpu/igpu0/prof rw,
+/dev/char/498:1 rw,
+/dev/char/498:2 rw,
+/dev/host1x-fence rw,
+/dev/shm/memmap_ipc_shm rw,
+
+/sys/module/firmware_class/parameters/path rw,
+`
+
+var nvidiaTegraDriversSupportConnectedPlugUdev = []string{
+	`KERNEL=="device-mapper", NAME="mapper/control"`,
+	`SUBSYSTEM!="block", GOTO="dm_end"`,
+	`KERNEL!="dm-[0-9]*", GOTO="dm_end"`,
+	`ACTION!="add|change", GOTO="dm_end"`,
+	`ENV{DM_UDEV_PRIMARY_SOURCE_FLAG}="1"`,
+	`LABEL="dm_end"`,
+
+	`ACTION=="add" SUBSYSTEM=="sdio" ATTR{vendor}=="0x02d0" RUN+="/etc/systemd/nvwifibt-pre.sh register $attr{device}"`,
```

Comment:
This looks like something that should be configured locally by the relevant gadget snap. I'm slightly confused why those rules are part of the interface. The rules here should provide means for identifying the devices such that a snap plugging the interface can access them and not set up random things on the platform.

The way this normally works for UC devices, is that there is a gadget which ships the necessary platform setup bits and access to locations such as /etc and so on is granted through system-files.